engine/engine.go

```diff
@@ -531,6 +531,13 @@ type compatibilityQuery struct {
 	scanners engstorage.Scanners
 }
 
+func (q *compatibilityQuery) totalSteps() int {
+	if q.step == 0 {
+		return 1
+	}
+	return int((q.end.UnixMilli()-q.start.UnixMilli())/q.step.Milliseconds() + 1)
```

Comment:
Don't we have this calculation in Options?

Reply:
Yeah it should probably live there but we don't have it yet.

Reply:
We have this: https://github.com/thanos-io/promql-engine/blob/main/query/options.go#L23-L38

Reply:
But I mean we can at least consolidate :)